test/e2e/network/dns_scale_records.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/kubernetes/test/e2e/framework"
+	testutils "k8s.io/kubernetes/test/utils"
+
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = SIGDescribe("PerformanceDNS [Feature:PerformanceDNS]", func() {
+	f := framework.NewDefaultFramework("performancedns")
+
+	BeforeEach(func() {
+		framework.ExpectNoError(framework.WaitForAllNodesSchedulable(f.ClientSet, framework.TestContext.NodeSchedulableTimeout))
+		framework.WaitForAllNodesHealthy(f.ClientSet, time.Minute)
+
+		err := framework.CheckTestingNSDeletedExcept(f.ClientSet, f.Namespace.Name)
+		framework.ExpectNoError(err)
+	})
+
+	// answers dns for service - creates the maximum number of services, and then check dns record for one
+	It("answers dns for service", func() {
+		services := generateDnsServicesInNamespace(f.Namespace.Name, 10000)
+		createService := func(i int) {
+			defer GinkgoRecover()
+			framework.ExpectNoError(testutils.CreateServiceWithRetries(f.ClientSet, f.Namespace.Name, services[i]))
+		}
+		workqueue.Parallelize(16, len(services), createService)
+		dnsTest := NewDnsTestCommon(f, f.Namespace.Name, "")
+		dnsTest.createUtilPodLabel("e2e-dns-scale-records")
+		defer dnsTest.deleteUtilPod()
+		s := services[len(services)-1]
+		svc, err := f.ClientSet.CoreV1().Services(s.Namespace).Get(s.Name, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		dnsTest.checkDNSRecordFrom(
+			fmt.Sprintf("%v.%v.svc.cluster.local", s.Name, s.Namespace),
+			func(actual []string) bool { return len(actual) == 1 && actual[0] == svc.Spec.ClusterIP },
```

Comment:
Could you please move the `return len(actual) == 1 && actual[0] == svc.Spec.ClusterIP` part to separate line?